### PR TITLE
Delete local user progress on account deletion (#25255) (#25283)

### PR DESCRIPTION
**Pull request by nausharipov**
Delete local user progress on account deletion.

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] Mention the appropriate issue in your description (for example: `addresses #123`), if applicable. This will automatically add a link to the pull request in the issue. If you would like the issue to automatically close on merging the pull request, comment `fixes #<ISSUE NUMBER>` instead.
 - [ ] Update `CHANGES.md` with noteworthy changes.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://beam.apache.org/contribute/get-started-contributing/#make-the-reviewers-job-easier).

To check the build health, please visit [https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md](https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md)

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)
[Go tests](https://github.com/apache/beam/actions?query=workflow%3A%22Go+tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI.

### CODE REVIEW

File: learning/tour-of-beam/frontend/lib/auth/notifier.dart [hunk 1]
```diff
@@ -54,7 +56,8 @@ class AuthNotifier extends ChangeNotifier {
 
   Future<void> deleteAccount() async {
     try {
-      await client.postDeleteUserProgress();
+      // TODO(nausharipov) review: is there a better place for this?
```
Comment (alexeyinkin) [inline learning/tour-of-beam/frontend/lib/auth/notifier.dart:59]: Add `AccountDeletionListener`, notify all listeners here.
Add `final _accountDeletionListeners = <AccountDeletionListener>[];`

With methods like `beforeAccountDeleted` or `afterAccountDeleted`. Likely we should do "before...". Those methods should be `async`, we should wait for them to complete.

File: learning/tour-of-beam/frontend/lib/cache/unit_progress.dart [hunk 2]
```diff
@@ -172,4 +172,9 @@ class UnitProgressCache extends ChangeNotifier {
       unitId: unitId,
     );
   }
+
+  Future<void> deleteUserProgress() async {
+    await _localStorageUserProgressRepository.deleteUserProgress();
+    await _cloudUserProgressRepository.deleteUserProgress();
```
Comment (alexeyinkin) [inline learning/tour-of-beam/frontend/lib/cache/unit_progress.dart:178]: `await Future.wait([...]);`

**Review by alexeyinkin — CHANGES_REQUESTED** [review]

File: learning/tour-of-beam/frontend/lib/auth/notifier.dart (hunk 1 above)
Comment (nausharipov) [reply learning/tour-of-beam/frontend/lib/auth/notifier.dart:59]: The implementation added extra code, which is not necessary for now. Added a comment.

**Review by alexeyinkin — APPROVED** [review]
LGTM (Internal review).

**Pull request merged**